mozregression/bisector.py

```diff
@@ -255,16 +288,20 @@ def bisect_nightlies(self, good_date, bad_date):
                                   " sure no inbound revision is missed)")
                 infos = {}
                 days = 6
+                first_date = min(handler.good_date, handler.bad_date)
                 while not 'changeset' in infos:
                     days += 1
-                    prev_date = handler.good_date - datetime.timedelta(days=days)
+                    prev_date = first_date - datetime.timedelta(days=days)
                     infos = handler.build_data.get_build_infos_for_date(prev_date)
                 if days > 7:
                     self._logger.info("At least one build folder was"
                                       " invalid, we have to start from"
                                       " %d days ago." % days)
-                return self.bisect_inbound(infos['changeset'],
-                                           handler.bad_revision)
+                good_rev, bad_rev = infos['changeset'], handler.bad_revision
+                if handler.hunt_fix:
+                    good_rev = handler.good_revision
+                    bad_rev = infos['changeset']
```

Comment:
May as well squeeze these two statements into good_rev, bad_rev = handle.good_revision, infos['changeset']. Also, I would put "good_rev, bad_rev = infos['changeset'], handler.bad_revision" in an else clause.